src/Foundatio/Storage/FolderFileStorage.cs

```diff
@@ -88,29 +88,19 @@ public class FolderFileStorage : IFileStorage {
             path = path.NormalizePath();
 
             string file = Path.Combine(Folder, path);
+            string directory = Path.GetDirectoryName(file);
+
+            Directory.CreateDirectory(directory);
```

Comment:
We don't want to call this unless there is an exception on File.Create, this has overhead and we don't need it in most cases